client/js/app/utils/ExplorerUtils.js

```diff
@@ -210,32 +207,74 @@ module.exports = {
     return params;
   },
 
-  getChartTypeOptions: function(response, analysisType) {
-    var chartTypes = [];
+  getQueryDataType: function(query){
+    var isInterval = typeof query.interval === "string",
+    isGroupBy = typeof query.group_by === "string",
+    is2xGroupBy = query.group_by instanceof Array,
+    dataType;
 
-    if (response) {
-      var dataviz = new Keen.Dataviz();
-      dataviz.data(response);
-      var dataType = dataviz.dataType();
+    // metric
+    if (!isGroupBy && !isInterval) {
+      dataType = 'singular';
+    }
 
-      if (dataType && Keen.Dataviz.dataTypeMap[dataType]) {
-        var library = Keen.Dataviz.dataTypeMap[dataType].library;
-        var libraryDefaults = Keen.Dataviz.libraries[library]._defaults;
-        chartTypes = _.clone(libraryDefaults[dataType]);
+    // group_by, no interval
+    if (isGroupBy && !isInterval) {
+      dataType = 'categorical';
+    }
 
-        if (!_.contains(chartTypes, 'json')) {
-          chartTypes.push('JSON');
-        }
-      } else if (response && _.contains(['extraction', 'select_unique'], analysisType)) {
-        chartTypes = ['JSON', 'table'];
-      }
+    // interval, no group_by
+    if (isInterval && !isGroupBy) {
+      dataType = 'chronological';
+    }
+
+    // interval, group_by
+    if (isInterval && isGroupBy) {
+      dataType = 'cat-chronological';
+    }
+
+    // 2x group_by
+    // TODO: research possible dataType options
+    if (!isInterval && is2xGroupBy) {
+      dataType = 'categorical';
     }
 
-    return chartTypes;
+    // interval, 2x group_by
+    // TODO: research possible dataType options
+    if (isInterval && is2xGroupBy) {
+      dataType = 'cat-chronological';
+    }
+
+    if (query.analysis_type === "funnel") {
+      dataType = 'cat-ordinal';
+    }
+
+    if (query.analysis_type === "extraction") {
+      dataType = 'extraction';
+    }
+    if (query.analysis_type === "select_unique") {
+      dataType = 'nominal';
+    }
+
+    return dataType;
+  },
+
+  getChartTypeOptions: function(query) {
+    var dataTypes = {
+      'categorical':        ['piechart', 'barchart', 'columnchart', 'table'],
```

Comment:
it looks like `singular` is missing from the `dataTypes` object in `getChartTypeOptions` in ExporerUtils